os/arch/xtensa/src/xtensa/xtensa_checkstack.c

```diff
@@ -0,0 +1,223 @@
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+
+#include <sys/types.h>
+#include <stdint.h>
+#include <sched.h>
+#include <assert.h>
+#include <debug.h>
+
+#include <tinyara/arch.h>
+#include <tinyara/tls.h>
+#include <tinyara/board.h>
+
+#include "xtensa.h"
+#include "sched/sched.h"
+
+#ifdef CONFIG_STACK_COLORATION
+
+/****************************************************************************
+ * Private Function Prototypes
+ ****************************************************************************/
+
+static size_t do_stackcheck(uintptr_t alloc, size_t size);
+
+/****************************************************************************
+ * Name: do_stackcheck
+ *
+ * Description:
+ *   Determine (approximately) how much stack has been used be searching the
+ *   stack memory for a high water mark.  That is, the deepest level of the
+ *   stack that clobbered some recognizable marker in the stack memory.
+ *
+ * Input Parameters:
+ *   alloc - Allocation base address of the stack
+ *   size - The size of the stack in bytes
+ *
+ * Returned Value:
+ *   The estimated amount of stack space used.
+ *
+ ****************************************************************************/
+
+static size_t do_stackcheck(uintptr_t alloc, size_t size)
+{
+	FAR uintptr_t start;
+	FAR uintptr_t end;
+	FAR uint32_t *ptr;
+	size_t mark;
+
+	if (size == 0) {
+		return 0;
+	}
+
+	/* Get aligned addresses of the top and bottom of the stack */
+
+#ifdef CONFIG_TLS
+	/* Skip over the TLS data structure at the bottom of the stack */
+
+	DEBUGASSERT((alloc & TLS_STACK_MASK) == 0);
+	start = alloc + sizeof(struct tls_info_s);
+#else
+	start = alloc & ~3;
+#endif
+	end = (alloc + size + 3) & ~3;
+
+	/* Get the adjusted size based on the top and bottom of the stack */
+
+	size = end - start;
+
+	/* The Xtensa CPUs use a push-down stack:  the stack grows toward lower
+	 * addresses in memory.  We need to start at the lowest address in the
+	 * stack memory allocation and search to higher addresses.  The first word
+	 * we encounter that does not have the magic value is the high water mark.
+	 */
+
+	for (ptr = (FAR uint32_t *) start, mark = (size >> 2); *ptr == STACK_COLOR && mark > 0; ptr++, mark--) ;
```

Comment:
Remove space between typecasting and variable.

Reply:
Fixed, thanks.